File: svc-data/src/test/java/com/beancounter/marketdata/FigiSerializationTest.java
package com.beancounter.marketdata;

import static org.assertj.core.api.Assertions.assertThat;

import com.beancounter.common.utils.BcJson;
import com.beancounter.marketdata.assets.figi.FigiResponse;
import com.fasterxml.jackson.core.type.TypeReference;
import java.util.Collection;
import org.junit.jupiter.api.Test;
import org.springframework.core.io.ClassPathResource;

public class FigiSerializationTest {
  @Test
  void is_ResultSerializable() throws Exception {
    Collection<FigiResponse> responses = new BcJson().getObjectMapper().readValue(
        new ClassPathResource("/contracts" + "/figi/multi-asset-response.json").getFile(),
        new TypeReference<>() {
        });

    assertThat(responses).hasSize(2);
  }

}

File: svc-data/src/test/kotlin/com/beancounter/marketdata/FigiSerializationTest.kt
package com.beancounter.marketdata

import com.beancounter.common.utils.BcJson
import com.beancounter.marketdata.assets.figi.FigiResponse
import com.fasterxml.jackson.core.type.TypeReference
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.core.io.ClassPathResource

/**
 * Mocked Figi responses serialize
 */
class FigiSerializationTest {
    @Test
    @Throws(Exception::class)
    fun figiResponseSerializable() {
        val responses: Collection<FigiResponse> = BcJson().objectMapper.readValue(
            ClassPathResource("/mock/figi/multi-asset-response.json").file,
            object : TypeReference<Collection<FigiResponse>>() {}
        )
        Assertions.assertThat(responses).hasSize(2)
    }
}
